submissions/UA-girl/task_friends_app/js/app.js

```diff
@@ -0,0 +1,354 @@
+        }
+        const fragment = document.createDocumentFragment();
+        cardsRender.forEach(person => fragment.append(person.card));
+        this.container.appendChild(fragment);
+        this.disableEnablePageToggler();
+    }
+
+    addListenersFilters() {
+        const app = this;
+        filterSet.addEventListener('input', function (event) {
+            const target = event.target.closest('input');
+            if (!target) {
+                return
+            }
+            const name = target.getAttribute('name');
+            const value = target.value;
+            switch (name) {
+                case 'search':
+                    app.nameFilter(value);
+                    break;
+                case 'sorting':
+                    app.sortCards(value, target.dataset.name);
+                    break;
+                case 'gender':
+                    const wasToggled = app.filterStates.gender.toggled;
+                    app.genderFilter(value, wasToggled);
+                    break;
+                case 'range':
+                    app.ageFilter(value);
+                    break;
+            }
+            app.currentPage = 0;
+            app.displayPeople();
+        });
+
+        filterSet.addEventListener('click', function ({target}) {
+            target = target.closest('input.button-reset');
+            if (!target) {
+                return
+            }
+            app.currentPage = 0;
+            app.resetApp();
+            app.displayPeople();
+        })
+    }
+
+    addEventListenersPages() {
+        const app = this;
+        pages.addEventListener('click', function({target}) {
+            const t = target.closest('button.button-page');
+            if (!t) {
+                return
+            }
+            if (t.classList.contains('button-page__back')){
+                app.currentPage = app.currentPage - 1;
+            } else if(t.classList.contains('button-page__forward')){
+                app.currentPage = app.currentPage + 1;
+            }
+            app.disableEnablePageToggler();
+            app.displayPeople();
+        });
+    }
+}
+
+
+function startApp(peopleData) {
+    const app = new App(document.querySelector('.cards'));
+    createPeopleList(peopleData);
+    app.cards = cards.slice();
+    removeOverlay();
+    app.displayPeople();
+    app.addListenersFilters();
+    app.addEventListenersPages();
+}
+
+function removeOverlay() {
+    const o = document.querySelector('.overlay');
+    o.classList.add('fade');
+}
+
+function createPeopleList(data) {
+    data.forEach(function (person) {
+        const {name, dob: {age}, gender, location: {country, city}, email, phone, cell, picture, registered} = person;
+        let newPerson = new Person(name, age, gender, country, city, email, phone, cell, picture, registered);
+        newPerson.cardTemplate();
+        cards.push(newPerson);
+    });
+}
+
+fetch(URL_FETCH)
+    .then(response => {
```

Comment:
You can simplify this expression:
```suggestion
    .then(response => response.json())
```